### PR TITLE
feat: migrate react-intl to lingui

### PR DESCRIPTION
**Pull request by rajat-ws**
### Ticket Link

---

### Related Links

---

### Description

- Migration of `react-intl` to `lingui`
- Reduced bundle size
- Updated UI

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

Before:

Screenshot 2024-07-23 at 1 16 24 PM

Screenshot 2024-07-23 at 1 16 35 PM

Screenshot 2024-07-23 at 1 22 53 PM

After:

Screenshot 2024-07-23 at 12 53 18 PM

Screenshot 2024-07-23 at 12 53 32 PM

Screenshot 2024-07-23 at 1 23 04 PM

- Updated after removing `formatjs` dependency:

Screenshot 2024-08-19 at 4 18 37 PM

### CODE REVIEW

**Review by praveenkumar1798 — CHANGES_REQUESTED** [review]
@rajat-ws There is a format js asset that is consuming about 1.3 Mb which is ideally not required. I have fixed it in react template. please do refer.

image

**Comment by rajat-ws** [comment]
> @rajat-ws There is a format js asset that is consuming about 1.3 Mb which is ideally not required. I have fixed it in react template. please do refer.
> 
> image

Removed it.

File: app/themes/colors.js [hunk 1]
```diff
@@ -5,14 +5,16 @@
...
+const lightGreen = '#607274';
+const text = '#4D3D0C';
 const secondaryText = '#FFFFFF';
+const tertiaryText = '#4D3D0C';
 const secondary = '#b0b0b0';
 const success = '#28a745';
 const error = '#dc3545';
 const gotoStories = '#1890ff';
-const cardBg = '#d8d8d8';
+const cardBg = '#EFF1F3';
```
Comment (praveenkumar1798) [inline app/themes/colors.js:17]: Why are these changes there ?

File: app/containers/LanguageProvider/index.tsx [hunk 2]
```diff
@@ -10,21 +10,22 @@ import React, { PropsWithChildren } from 'react';
...
 export function LanguageProvider({ locale, messages, children }: PropsWithChildren<LanguageProviderProps>) {
-  return (
-    <IntlProvider locale={locale!} key={locale} messages={messages[locale!]}>
-      <IntlGlobalProvider>{React.Children.only(children)}</IntlGlobalProvider>
-    </IntlProvider>
-  );
+  const localizedMessages = messages[locale!];
+
+  i18n.load(locale!, localizedMessages);
+  i18n.activate(locale!);
```
Comment (praveenkumar1798) [inline app/containers/LanguageProvider/index.tsx:26]: Why is it locale!. Is this expectedb ?

File: app/components/Siderbar/index.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ const SideBarStatic = styled.div`
     max-width: 7rem;
     min-height: calc(100vh - ${(props) => props.theme.headerHeight});
     height: auto;
-    background-color: ${colors.primary};
+    background-color: ${colors.lightGreen};
```
Comment (praveenkumar1798) [inline app/components/Siderbar/index.tsx:38]: Why have we changed this ?

File: app/components/Header/index.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ const StyledHeader = styled(Layout.Header)`
       height: ${(props) => props.theme.headerHeight};
       align-items: center;
       justify-content: center;
-      background-color: ${colors.primary};
+      background-color: ${colors.lightGreen};
```
Comment (praveenkumar1798) [inline app/components/Header/index.tsx:21]: Why have changed this ? Any reason ?

File: app/themes/colors.js (hunk 1 above)
Comment (rajat-ws) [reply app/themes/colors.js:17]: Current ui of app looked somewhat awful hence I tried to modify it. Have added screenshots in the PR description for the reference.

File: app/components/Header/index.tsx (hunk 4 above)
Comment (rajat-ws) [reply app/components/Header/index.tsx:21]: Tried to make UI more appealing and hence made those changes. I have added screenshots in the PR description for the reference.

File: app/components/Siderbar/index.tsx (hunk 3 above)
Comment (rajat-ws) [reply app/components/Siderbar/index.tsx:38]: Tried to make UI more appealing and hence made those changes. I have added screenshots in the PR description for the reference.

File: app/containers/LanguageProvider/index.tsx (hunk 2 above)
Comment (rajat-ws) [reply app/containers/LanguageProvider/index.tsx]: Pushed the fix for it.

**Review by praveenkumar1798 — APPROVED** [review]

**Pull request merged**